File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic={}
        for i in strs:
            sortedstr=''.join(sorted(i))
            if sortedstr not in dic:
                dic[sortedstr]=[i]
            else:
                if sortedstr in dic:
                    dic[sortedstr].append(i)
        return list(dic.values())
        

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> res=new HashMap<>();
        for(int i=0;i<strs.length;i++){
            char[] chr=strs[i].toCharArray();
            Arrays.sort(chr);
            String ss=new String(chr);
            if(!res.containsKey(ss)){
                res.put(ss,new ArrayList<String>());
                res.get(ss).add(strs[i]);
            }
            else{
                res.get(ss).add(strs[i]);
            }
        }
        return new ArrayList<>(res.values());
    }
}